packages/xo-web/src/xo-app/new/network/index.js

```diff
@@ -0,0 +1,252 @@
+import _, { messages } from 'intl'
+import ActionButton from 'action-button'
+import BaseComponent from 'base-component'
+import map from 'lodash/map'
+import PropTypes from 'prop-types'
+import React from 'react'
+import Wizard, { Section } from 'wizard'
+import { Toggle } from 'form'
+import { injectIntl } from 'react-intl'
+import { Container, Row, Col } from 'grid'
+import { SelectPool, SelectPif } from 'select-objects'
+import { addSubscriptions, connectStore } from 'utils'
+import {
+  getBondModes,
+  createBondedNetwork,
+  createNetwork,
+  subscribeCurrentUser,
+} from 'xo'
+import {
+  createSelector,
+  createGetObject,
+  getIsPoolAdmin,
+  getUser,
+} from 'selectors'
+
+import styles from './index.css'
+import Page from '../../page'
+
+// =============================================================================
+
+@addSubscriptions({
+  user: subscribeCurrentUser,
+})
+@connectStore(() => ({
+  isAdmin: createSelector(
+    getUser,
+    user => user && user.permission === 'admin'
+  ),
+  isPoolAdmin: getIsPoolAdmin,
+  pool: createGetObject((_, props) => props.location.query.pool),
+}))
+@injectIntl
+export default class NewNetwork extends BaseComponent {
+  static contextTypes = {
+    router: PropTypes.object,
+  }
+
+  state = {
+    bondedNetwork: false,
+  }
+
+  componentWillMount() {
+    getBondModes().then(bondModes =>
+      this.setState({ bondModes, bondMode: bondModes[0] })
+    )
+  }
+
+  _reset = () => {
+    this.setState({
+      bondedNetwork: false,
+      description: '',
+      mtu: '',
+      name: '',
+      pif: undefined,
+      pifs: undefined,
+      vlan: '',
+    })
+  }
+
+  _create = () => {
+    const {
+      bondedNetwork,
+      bondMode,
+      description,
+      mtu,
+      name,
+      pif,
+      pifs,
+      vlan,
+    } = this.state
+    return bondedNetwork
+      ? createBondedNetwork({
+          bondMode,
+          description,
+          mtu,
+          name,
+          pifs: map(pifs, pif => pif.id),
+          pool: this.props.pool.id,
+          vlan,
+        })
+      : createNetwork({
+          description,
+          mtu,
+          name,
+          pif: pif.id,
+          pool: this.props.pool.id,
+          vlan,
+        })
+  }
+
+  _getPifPredicate = createSelector(
+    () => this.props.pool,
+    pool => pif => pif.vlan === -1 && pif.$host === (pool && pool.master)
+  )
+
+  _getRedirectionUrl = network =>
+    `pools/${this.props.pool.id}/network?s=${network.uuid}`
```

Comment:
Do we really want to filter the network list when we go back to the pool view?

Reply:
For the user to see that the network is created.

Reply:
Yes, I get it but I think it's a bit weird that you only see one network when you come back from creating a new network. IMO, it should pretty much behave like the modal we had before except it's not a modal anymore: Click "Add network" > Fill the form > Click OK > The network appears in the list among the other ones.